corehq/messaging/smsbackends/telerivet/models.py

```diff
@@ -118,11 +123,11 @@ def by_webhook_secret(cls, webhook_secret):
         result = cls.active_objects.filter(
             hq_api_id=cls.get_api_id()
         )
-        result_by_webhook = {
-            backend.config.webhook_secret: backend
-            for backend in result
-        }
-        return result_by_webhook.get(webhook_secret)
+
+        for backend in result:
+            if backend.config.webhook_secret == webhook_secret:
+                return backend
```

Comment:
Oh, nice catch.

Would it be better to raise an exception if two configurations have the same secret? That does sound dangerous.

Reply:
thanks @kaapstorm